core/src/com/unciv/logic/map/mapunit/UnitPromotions.kt

```diff
@@ -48,11 +49,19 @@ class UnitPromotions : IsPartOfGameInfoSerialization {
     }
 
     /** @return the XP points needed to "buy" the next promotion. 10, 30, 60, 100, 150,... */
-    fun xpForNextPromotion() = (numberOfPromotions + 1) * 10
+    fun xpForNextPromotion() = costOfPromotion((numberOfPromotions + 1) * 10)
     
     /** @return the XP points needed to "buy" the next [count] promotions. */
-    fun xpForNextNPromotions(count: Int) = (1..count).sumOf { (numberOfPromotions + it) * 10 }
-
+    fun xpForNextNPromotions(count: Int) = costOfPromotion((1..count).sumOf { (numberOfPromotions + it) * 10 })
+    
+    /** return the existingPromotions cost with the XPForPromotionModifier Unique applied */
+    private fun costOfPromotion(existingPromotions: Int): Int {
+        for (unique in unit.civ.getMatchingUniques(UniqueType.XPForPromotionModifier)) {
+            return Math.round((existingPromotions * unique.params[0].toPercent()))
```

Comment:
I'm pretty sure you don't want 'return' - of there are multiple uniques they should all reduce the cost, not just the first one

Reply:
Also remove extra parentheses